packages/@aws-cdk/aws-codebuild/test/test.project.ts

```diff
@@ -1231,6 +1231,110 @@ export = {
         test.done();
       },
 
+      "when provided as a Token referencing a different account, adds permission to decrypt keys in the Secret's account"(test: Test) {
+        // GIVEN
+        const stack = new cdk.Stack();
+        const secretToken = cdk.Lazy.string({ produce: () => 'arn:aws:secretsmanager:us-west-2:901234567890:secret:mysecret' });
+
+        // WHEN
+        const secret = secretsmanager.Secret.fromSecretPartialArn(stack, 'Secret', secretToken);
+        new codebuild.PipelineProject(stack, 'Project', {
+          environmentVariables: {
+            'ENV_VAR1': {
+              type: codebuild.BuildEnvironmentVariableType.SECRETS_MANAGER,
+              value: secret.secretArn,
+            },
+          },
+        });
+
+        // THEN
+        expect(stack).to(haveResourceLike('AWS::IAM::Policy', {
+          'PolicyDocument': {
+            'Statement': arrayWith({
+              'Action': 'kms:Decrypt',
+              'Effect': 'Allow',
+              'Resource': {
+                'Fn::Join': [
+                  '',
+                  [
+                    'arn:',
+                    { 'Ref': 'AWS::Partition' },
+                    ':kms:',
+                    { 'Fn::Select': [3, { 'Fn::Split': [':', 'arn:aws:secretsmanager:us-west-2:901234567890:secret:mysecret'] }] },
+                    ':',
+                    { 'Fn::Select': [4, { 'Fn::Split': [':', 'arn:aws:secretsmanager:us-west-2:901234567890:secret:mysecret'] }] },
+                    ':key/*',
+                  ],
+                ],
+              },
+            }),
+          },
+        }));
+
+        test.done();
+      },
+
+      'when the same new secret is provided with different JSON keys, only adds the resource once'(test: Test) {
+        // GIVEN
+        const stack = new cdk.Stack();
+
+        // WHEN
+        const secret1 = new secretsmanager.Secret(stack, 'Secret');
+        new codebuild.PipelineProject(stack, 'Project', {
```

Comment:
Here we have a secret that is provided as a token and is used twice with two different JSON keys. The test fails because we add the Ref to the resource twice.

Reply:
I actually think the test _will_ pass with the way it's written today, but I see your point 🙂.

Is this just a matter of changing to `Set` [here](https://github.com/aws/aws-cdk/blob/e31587a2d644637ba85449396c6274bcb5e1569a/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L723-L723)?

Reply:
The test actually fails locally when I run it, because the resource is added twice. I am open for suggestions though if it is not clear enough :)

Reply:
Ah, you're right (the same resource is present twice in the `Resource` property of resulting IAM statement). Good catch!

Reply:
Any chance of a separate PR fixing just this bug? 🙂

Reply:
Sure I can do that. Then let me rename this bug for that fix and I will create a separate one for the other one.